### ISSUE TITLE
Warn in fork mode with 1 instance may not be a warn.

### ISSUE THREAD

**Issue by poppinlp** [Unitech/pm2#2872]
**What's going wrong?**

Warn in fork mode with 1 instance may not be a warn since there's only 1 process.

**How could we reproduce this issue?**

1. Create a process file
2. Set `instances` to `1` and `exec_mode` to `fork`
3. Start it then you can see warn message:

> [PM2][WARN] You are starting 1 processes in fork_mode without load balancing. To enable it remove -x option.

If remove `instances` option in process file, the warn message is gone.
It's some kind of weird. Maybe missed a condition check before show this warn message?

## Supporting information

```
PM2 version: 4.2.0
Node version: 7.10.0
Mac
```

**Comment by poppinlp** [comment]
I find this [code](https://github.com/Unitech/pm2/blob/1c3fcb83b65e430a5734d576610d27cc7a0bf4fd/lib/Common.js#L598):

```js
!isNaN(conf.instances) && /^fork(_mode)?$/i.test(conf.exec_mode)
```

It leads the warn message for fork mode with any(not NAN) instances number.
I don't think it makes sense and be glad to send a PR to fix.
But if you think it's what it should be, then just ignore me. :)

**Comment by soyuka** [comment]
Hmm, yes the warning is on purpose here but maybe the message isn't clear enough? 

It's just warning you that `instances` has no impact on the `fork` mode. Indeed, this option is only valid with `cluster` and might be misleading.

**Comment by Unitech** [comment]
It's a bit of a edge case, but feel free to send a PR

**Comment by vmarchaud** [comment]
Published under PM2 `2.5`
